src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNILoadHandle.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading;
+
+namespace System.Data.SqlClient.SNI
+{
+    /// <summary>
+    /// Global SNI settings and status
+    /// </summary>
+    internal class SNILoadHandle
+    {
+        static public SNILoadHandle SingletonInstance = new SNILoadHandle();
+
+        private ThreadLocal<SNIError> _lastError = new ThreadLocal<SNIError>(() => { return new SNIError(SNIProviders.INVALID_PROV, 0, 0, "No SNI Error has been reported yet."); });
+        private readonly UInt32 _status = TdsEnums.SNI_SUCCESS;
```

Comment:
Nit: another example where it'd be preferable to use a C# keyword instead of type name, e.g. uint instead of UInt32.  I'll stop commenting on these.